### PR TITLE
3/doc more menu

### PR DESCRIPTION
**Pull request by stuartromanek**
kicks off https://apostrophecms.atlassian.net/browse/A30U-731?atlOrigin=eyJpIjoiMDM2ODNiYzE0YzE4NDNmYzgzYzE3YzU2MDYyNDNkN2MiLCJwIjoiaiJ9
- requires follow up tickets for all functionality ([871](https://apostrophecms.atlassian.net/browse/A30U-871), [872](https://apostrophecms.atlassian.net/browse/A30U-872), [873](https://apostrophecms.atlassian.net/browse/A30U-873),)
- Currently the new component comes ready to roll out the whole menu but maybe this should be provided by the parent context

### CODE REVIEW

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarUser.vue [hunk 1]
```diff
@@ -4,7 +4,6 @@
     :button="button"
     :menu="menu"
     menu-placement="bottom-end"
-    :menu-offset="2"
```
Comment (stuartromanek) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarUser.vue:7]: unrelated menu fiddling

File: modules/@apostrophecms/ui/ui/apos/components/AposContextMenu.vue [hunk 2]
```diff
@@ -6,6 +6,7 @@
       @show="show"
       :offset="menuOffset"
       trigger="manual"
+      :placement="menuPlacement"
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/components/AposContextMenu.vue:9]: unrelated menu fiddling

File: modules/@apostrophecms/ui/ui/apos/components/AposContextMenuItem.vue [hunk 3]
```diff
@@ -2,6 +2,7 @@
   <li class="apos-context-menu__item">
     <button
       class="apos-context-menu__button"
+      :class="modifiers"
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/components/AposContextMenuItem.vue:5]: new options for context menu items

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocMoreMenu.vue [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    return {
+      isOpen: false,
+      menu: [
+        {
+          label: 'Share Draft',
+          action: 'share'
+        },
+        {
+          label: 'Save Draft',
+          action: 'save'
```
Comment (boutell) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocMoreMenu.vue:39]: This action does not exist, we are autosaving.

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocMoreMenu.vue [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        },
+        {
+          label: 'Duplicate Document',
+          action: 'duplicate'
+        },
+        {
+          label: 'Version History',
+          action: 'versions'
```
Comment (boutell) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocMoreMenu.vue:47]: This does not exist, but we are going to have to make this list extensible by pushing it from the server via browserOptions soon (not a blocker for now).

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocMoreMenu.vue [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    },
+    async versions() {
+      await apos.modal.execute('AposDocVersions', {
+        options: { foo: 'bar' },
+        doc: {
+          _id: this.docId
+        }
+      });
+    },
+    menuHandler(action) {
```
Comment (abea) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocMoreMenu.vue:78]: Probably will need to be an async function, but also not necessary for this story.

**Review by abea — COMMENTED** [review]
The stuff Tom said, but otherwise it looks good.

**Review by boutell — APPROVED** [review]

**Pull request merged**